### DISCUSSION TITLE
Install Scripts

### DISCUSSION THREAD

**Discussion by hopoffacloud** [magma/magma#5608] [Ideas]
I have been chatting with staff in house. 
We are playing with the options for Magma. 

At present it does not look like there is an install script say for Ubuntu or Debian or whatever that allows the install of all dependencies etc. 

If this does exist - Awesome. 

If not - I am thinking our in house team may take this on. 

I am thinking the script should have a number of things where it pauses, asks questions and then continues. 

Such as - what is the ip for the SQL server, if none - installs one, asking for passwords etc that you would want to be used VS having them be generic and thus a security issue long term. 

Anyhow - figured I would ask here on the mailing list if this exists. 

Kind Regards, 

Glenn S. Kelley
I am a www.Connectivity.Engineer

**Comment by karthiksubraveti** [comment]
Adding @jeremy-donson and @pshelar. They might have better information here.

**Reply by hopoffacloud** [reply]
awesome. thank you.

We help a lot of wisps - both here in the States as well as worldwide. 
I know something like this would be valuable - 
I know dropping this into a chassis with either Vyatta, Mikrotik or another Virtual Router (cloud6 or similar) this would be an excellent box to drop into a number of networks for WISPs. 

Thus the hope to make a config script to help folks get started. . .

**Reply by karthiksubraveti** [reply]
Thanks for the context. FYI this track this effort 
https://app.zenhub.com/workspaces/magma--devops-5f98596bdbbaee000ef2c1f6/issues/magma/magma/5168

**Reply by mattymo** [reply]
@hopoffacloud Are you just interested in automation for the AGW? Or also for deploying orc8r + nms and Kubernetes itself? The AGW doesn't communicate directly with the DB server, so I'm not sure what direction you are looking for help with automation.

**Comment by arunuke** [comment]
If the objective is to get a Ubuntu node with all dependencies needed to build or deploy Magma, please check out the code that landed a few days back which configures a build node with all the necessary dependencies from https://github.com/magma/magma/blob/master/experimental/cloudstrapper/playbooks/build-configure.yaml. There is a higher level README that talks about how to easily build Magma on Ubuntu (either on a localhost or an image from a Public Cloud provider) and how to deploy Orchestrator on AWS available from https://github.com/magma/magma/tree/master/experimental/cloudstrapper

**Comment by hopoffacloud** [comment]
perfect.

My hope is to make a script that most any MSO / Wisp can use that 
basically asks simple questions

Such as What is your PLMN, what network are you pushing to the UE's,   etc.

Thank fires up the build script.

I am on the road but will review this shortly



Thanks for the update

**Reply by arunuke** [reply]
I wanted to add a quick note to this based on the clarification above. The features above (PLMN, networks etc.) would be a great add-on to the existing automation which sets up Magma (the code I referred to above). Once the install is complete and a post check is run to determine all services are up, the script that automates an actual deployment would be a good value add for customers planning to get started.
